src/strategies/stablexswap/index.ts

```diff
@@ -0,0 +1,157 @@
+import { formatUnits } from '@ethersproject/units';
+import { multicall } from '../../utils';
+
+export const author = 'stablexswap';
+export const version = '0.0.1';
+
+const bep20abi = [
+  {
+    inputs: [
+      {
+        internalType: 'address',
+        name: '',
+        type: 'address'
+      }
+    ],
+    name: 'balanceOf',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: 'amount',
+        type: 'uint256'
+      }
+    ],
+    stateMutability: 'view',
+    type: 'function'
+  }
+];
+
+const masterChefAbi = [
+  {
+    inputs: [
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256'
+      },
+      {
+        internalType: 'address',
+        name: '',
+        type: 'address'
+      }
+    ],
+    name: 'userInfo',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: 'amount',
+        type: 'uint256'
+      },
+      {
+        internalType: 'uint256',
+        name: 'rewardDebt',
+        type: 'uint256'
+      }
+    ],
+    stateMutability: 'view',
+    type: 'function'
+  }
+];
+
+const stakingChefAbi = [
+  {
+    inputs: [
+      {
+        internalType: 'address',
+        name: '',
+        type: 'address'
+      }
+    ],
+    name: 'poolsInfo',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: 'amount',
+        type: 'uint256'
+      }
+    ],
+    stateMutability: 'view',
+    type: 'function'
+  }
+]
+
+const staxERC20ContractAddress = '0x0da6ed8b13214ff28e9ca979dd37439e8a88f6c4';
+const masterChefContractAddress = '0xc80991f9106e26e43bf1c07c764829a85f294c71';
+const stakingChefContractAddress = '0x0c0c475e32212b748c328e451ab3862ffe07369e';
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  // For erc20 token balances not staked in any contract
+  const score = await multicall(
```

Comment:
There is too much multicall in your strategy, this will take too long to resolve on the browser, you should use a single call instead, i recommend you to use "Multicaller" instead of the function "multicall()`, it's the new way we use to do multicall, it's much easier to build query with it, have a look on a demo here: https://github.com/snapshot-labs/snapshot.js/blob/master/test/multicaller.ts and a strategy using it here https://github.com/snapshot-labs/snapshot.js/blob/master/src/strategies/dittomoney/index.ts#L63-L82 to understand how it works.

Reply:
Appreciate the time taken to review . It makes a lot of sense to batch the calls into a single multicall - I've upgraded the design to use the Multicall class instead

Reply:
Looks good now thanks!